In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
spotify = pd.read_csv("/content/spotify_tracks.csv.csv")

In [5]:
spotify

,Unnamed: 0,track_id,artists,album_name,track_name,popularity,duration_ms,explicit,danceability,energy,...,loudness,mode,speechiness,acousticness,instrumentalness,liveness,valence,tempo,time_signature,track_genre
0,0,5SuOikwiRyPMVoIQDJUgSV,Gen Hoshino,Comedy,Comedy,73,230666,False,0.676,0.4610,...,-6.746,0,0.1430,0.0322,0.000001,0.3580,0.7150,87.917,4,acoustic
1,1,4qPNDBW1i3p13qLCt0Ki3A,Ben Woodward,Ghost (Acoustic),Ghost - Acoustic,55,149610,False,0.420,0.1660,...,-17.235,1,0.0763,0.9240,0.000006,0.1010,0.2670,77.489,4,acoustic
2,2,1iJBSr7s7jYXzM8EGcbK5b,Ingrid Michaelson;ZAYN,To Begin Again,To Begin Again,57,210826,False,0.438,0.3590,...,-9.734,1,0.0557,0.2100,0.000000,0.1170,0.1200,76.332,4,acoustic
3,3,6lfxq3CG4xtTiEg7opyCyx,Kina Grannis,Crazy Rich Asians (Original Motion Picture Sou...,Can't Help Falling In Love,71,201933,False,0.266,0.0596,...,-18.515,1,0.0363,0.9050,0.000071,0.1320,0.1430,181.740,3,acoustic
4,4,5vjLSffimiIP26QG5WcN2K,Chord Overstreet,Hold On,Hold On,82,198853,False,0.618,0.4430,...,-9.681,1,0.0526,0.4690,0.000000,0.0829,0.1670,119.949,4,acoustic
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
113995,113995,2C3TZjDRiAzdyViavDJ217,Rainy Lullaby,#mindfulness - Soft Rain for Mindful Meditatio...,Sleep My Little Boy,21,384999,False,0.172,0.2350,...,-16.393,1,0.0422,0.6400,0.928000,0.0863,0.0339,125.995,5,world-music
113996,113996,1hIz5L4IB9hN3WRYPOCGPw,Rainy Lullaby,#mindfulness - Soft Rain for Mindful Meditatio...,Water Into Light,22,385000,False,0.174,0.1170,...,-18.318,0,0.0401,0.9940,0.976000,0.1050,0.0350,85.239,4,world-music
113997,113997,6x8ZfSoqDjuNa5SVP5QjvX,Cesária Evora,Best Of,Miss Perfumado,22,271466,False,0.629,0.3290,...,-10.895,0,0.0420,0.8670,0.000000,0.0839,0.7430,132.378,4,world-music
113998,113998,2e6sXL2bYv4bSz6VTdnfLs,Michael W. Smith,Change Your World,Friends,41,283893,False,0.587,0.5060,...,-10.889,1,0.0297,0.3810,0.000000,0.2700,0.4130,135.960,4,world-music


In [6]:
spotify.head(10)

,Unnamed: 0,track_id,artists,album_name,track_name,popularity,duration_ms,explicit,danceability,energy,...,loudness,mode,speechiness,acousticness,instrumentalness,liveness,valence,tempo,time_signature,track_genre
0,0,5SuOikwiRyPMVoIQDJUgSV,Gen Hoshino,Comedy,Comedy,73,230666,False,0.676,0.4610,...,-6.746,0,0.1430,0.0322,0.000001,0.3580,0.7150,87.917,4,acoustic
1,1,4qPNDBW1i3p13qLCt0Ki3A,Ben Woodward,Ghost (Acoustic),Ghost - Acoustic,55,149610,False,0.420,0.1660,...,-17.235,1,0.0763,0.9240,0.000006,0.1010,0.2670,77.489,4,acoustic
2,2,1iJBSr7s7jYXzM8EGcbK5b,Ingrid Michaelson;ZAYN,To Begin Again,To Begin Again,57,210826,False,0.438,0.3590,...,-9.734,1,0.0557,0.2100,0.000000,0.1170,0.1200,76.332,4,acoustic
3,3,6lfxq3CG4xtTiEg7opyCyx,Kina Grannis,Crazy Rich Asians (Original Motion Picture Sou...,Can't Help Falling In Love,71,201933,False,0.266,0.0596,...,-18.515,1,0.0363,0.9050,0.000071,0.1320,0.1430,181.740,3,acoustic
4,4,5vjLSffimiIP26QG5WcN2K,Chord Overstreet,Hold On,Hold On,82,198853,False,0.618,0.4430,...,-9.681,1,0.0526,0.4690,0.000000,0.0829,0.1670,119.949,4,acoustic
5,5,01MVOl9KtVTNfFiBU9I7dc,Tyrone Wells,Days I Will Remember,Days I Will Remember,58,214240,False,0.688,0.4810,...,-8.807,1,0.1050,0.2890,0.000000,0.1890,0.6660,98.017,4,acoustic
6,6,6Vc5wAMmXdKIAM7WUoEb7N,A Great Big World;Christina Aguilera,Is There Anybody Out There?,Say Something,74,229400,False,0.407,0.1470,...,-8.822,1,0.0355,0.8570,0.000003,0.0913,0.0765,141.284,3,acoustic
7,7,1EzrEOXmMH3G43AXT1y7pA,Jason Mraz,We Sing. We Dance. We Steal Things.,I'm Yours,80,242946,False,0.703,0.4440,...,-9.331,1,0.0417,0.5590,0.000000,0.0973,0.7120,150.960,4,acoustic
8,8,0IktbUcnAGrvD03AWnz3Q8,Jason Mraz;Colbie Caillat,We Sing. We Dance. We Steal Things.,Lucky,74,189613,False,0.625,0.4140,...,-8.700,1,0.0369,0.2940,0.000000,0.1510,0.6690,130.088,4,acoustic
9,9,7k9GuJYLp2AzqokyEdwEw2,Ross Copperman,Hunger,Hunger,56,205594,False,0.442,0.6320,...,-6.770,1,0.0295,0.4260,0.004190,0.0735,0.1960,78.899,4,acoustic


In [9]:
spotify.columns

Index(['Unnamed: 0', 'track_id', 'artists', 'album_name', 'track_name',
       'popularity', 'duration_ms', 'explicit', 'danceability', 'energy',
       'key', 'loudness', 'mode', 'speechiness', 'acousticness',
       'instrumentalness', 'liveness', 'valence', 'tempo', 'time_signature',
       'track_genre'],
      dtype='object')

In [10]:
spotify.isnull().sum()

,0
Unnamed: 0,0
track_id,0
artists,1
album_name,1
track_name,1
popularity,0
duration_ms,0
explicit,0
danceability,0
energy,0


In [33]:
MY_GENRE = "world-music"
df_genre = spotify[spotify['track_genre'] == MY_GENRE].copy()
print(f"Tracks found for genre '{MY_GENRE}': {len(df_genre)}")

Tracks found for genre 'world-music': 1000


In [34]:
columns_to_drop = ['track_id', 'artists', 'album_name', 'track_name', 'Unnamed: 0', 'explicit', 'key', 'mode', 'time_signature']
existing_drops = [col for col in columns_to_drop if col in df_genre.columns]

In [35]:
df_genre.drop(columns=existing_drops, inplace=True, errors='ignore')
print(f"\nColumns remaining: {df_genre.columns.tolist()}")


Columns remaining: ['popularity', 'duration_ms', 'danceability', 'energy', 'loudness', 'speechiness', 'acousticness', 'instrumentalness', 'liveness', 'valence', 'tempo', 'track_genre']


In [36]:
print("\n--- Data Types & Nulls ---")
print(df_genre.info())



--- Data Types & Nulls ---
<class 'pandas.core.frame.DataFrame'>
Index: 1000 entries, 113000 to 113999
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   popularity        1000 non-null   int64  
 1   duration_ms       1000 non-null   int64  
 2   danceability      1000 non-null   float64
 3   energy            1000 non-null   float64
 4   loudness          1000 non-null   float64
 5   speechiness       1000 non-null   float64
 6   acousticness      1000 non-null   float64
 7   instrumentalness  1000 non-null   float64
 8   liveness          1000 non-null   float64
 9   valence           1000 non-null   float64
 10  tempo             1000 non-null   float64
 11  track_genre       1000 non-null   object 
dtypes: float64(9), int64(2), object(1)
memory usage: 101.6+ KB
None


In [37]:
print(df_genre.describe())

        popularity   duration_ms  danceability       energy     loudness  \
count  1000.000000  1.000000e+03   1000.000000  1000.000000  1000.000000   
mean     41.873000  2.971956e+05      0.414572     0.532987    -9.417547   
std      12.178472  1.455780e+05      0.136892     0.244229     6.013575   
min       0.000000  6.256300e+04      0.000000     0.000020   -43.043000   
25%      37.000000  2.305948e+05      0.329000     0.356250   -10.265000   
50%      44.000000  2.763600e+05      0.423500     0.550500    -7.626000   
75%      50.000000  3.394832e+05      0.509000     0.720000    -5.959000   
max      70.000000  3.601693e+06      0.838000     0.996000    -2.757000   

       speechiness  acousticness  instrumentalness     liveness      valence  \
count  1000.000000   1000.000000       1000.000000  1000.000000  1000.000000   
mean      0.041768      0.299547          0.091824     0.250393     0.250295   
std       0.034653      0.343136          0.270520     0.242990     0.17333

In [43]:
df_genre.to_csv('world-music_filtered.csv', index=False)

In [45]:
from google.colab import files
files.download('world-music_filtered.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [38]:
global_means = spotify[['energy', 'danceability', 'acousticness', 'valence', 'instrumentalness', 'speechiness', 'popularity']].mean()
genre_means = df_genre[['energy', 'danceability', 'acousticness', 'valence', 'instrumentalness', 'speechiness', 'popularity']].mean()


In [41]:
print("="*60)
print("MY WORLD-MUSIC NUMBERS")
print("="*60)
print(f"1. Global Avg Acousticness: {global_means['acousticness']:.3f}")
print(f"2. World-Music Avg Acousticness: {genre_means['acousticness']:.3f}")
print(f"3. World-Music Avg Energy: {genre_means['energy']:.3f} (Global: {global_means['energy']:.3f})")
print(f"4. World-Music Avg Danceability: {genre_means['danceability']:.3f} (Global: {global_means['danceability']:.3f})")
print(f"5. Instrumentalness - 75th Percentile: {df_genre['instrumentalness'].quantile(0.75):.3f}")
print(f"6. Instrumentalness - Median: {df_genre['instrumentalness'].median():.3f}")
print(f"7. World-Music Popularity - Mean: {genre_means['popularity']:.1f}")
print(f"8. World-Music Popularity - Std Dev: {df_genre['popularity'].std():.1f}")
print(f"9. World-Music Popularity - 90th Percentile: {df_genre['popularity'].quantile(0.90):.1f}")
print(f"10. Correlation (Energy vs Valence): {df_genre['energy'].corr(df_genre['valence']):.2f}")

MY WORLD-MUSIC NUMBERS
1. Global Avg Acousticness: 0.315
2. World-Music Avg Acousticness: 0.300
3. World-Music Avg Energy: 0.533 (Global: 0.641)
4. World-Music Avg Danceability: 0.415 (Global: 0.567)
5. Instrumentalness - 75th Percentile: 0.000
6. Instrumentalness - Median: 0.000
7. World-Music Popularity - Mean: 41.9
8. World-Music Popularity - Std Dev: 12.2
9. World-Music Popularity - 90th Percentile: 55.0
10. Correlation (Energy vs Valence): 0.42


## ***Observation 1 (Lower Energy & Danceability, Slightly Less Acoustic)***

Compared to the global average, world-music has slightly lower acousticness (0.300 vs global 0.315), but significantly lower energy (0.533 vs global 0.641) and danceability (0.415 vs global 0.567). This means world-music is not defined by raw acoustic instrumentation, but rather by a chill, downtempo, non-aggressive sound profile. It is background listening music, not high-energy dancefloor filler. This tells the streaming service that world-music listeners are likely using it for relaxation or focus, not parties.

# Observation 2 (Totally Vocal – No Instrumental Tracks)

Critically, both the median and the 75th percentile of instrumentalness are exactly 0.000. This means every single track in the world-music dataset contains significant vocals. There is no purely instrumental folk or ceremonial music in this data. The genre is overwhelmingly driven by vocal storytelling and singing. Playlists should be organized by vocal style or language, rather than separating vocal from instrumental – because instrumental simply does not exist in this dataset.

# **Observation** 3 (The "No Hits" Ceiling) *italicized text*

The mean popularity is 41.9 out of 100, with a standard deviation of only 12.2. The 90th percentile maxes out at 55.0. This means there are zero viral mega-hits in world-music. Even the best-performing tracks barely cross the halfway mark on Spotify's popularity scale. For Week 6's classification model, we must define a 'hit' with a low threshold (e.g., popularity > 50), and we will have very few positive examples to train on. This model will struggle with precision and recall, and we must explicitly tell the business team that identifying 'hits' in niche genres is inherently difficult without external marketing data.